Firestore/core/src/firebase/firestore/model/field_value.cc

```diff
@@ -153,6 +153,72 @@ FieldValue& FieldValue::operator=(FieldValue&& value) {
   }
 }
 
+FieldValue FieldValue::Set(FieldPath field_path, FieldValue value) const {
+  FIREBASE_ASSERT_MESSAGE(type() == Type::Object,
+                          "Cannot set field for non-object FieldValue");
+  FIREBASE_ASSERT_MESSAGE(!field_path.empty(),
```

Comment:
Should `Get` have a similar assertion?

Reply:
(avoiding the question a bit)

I don't think it's strictly necessary. If you call `FieldValue::Get("")`, then you essentially just return `this` which doesn't seem completely unreasonable. (Or at least, that's what this and the other ports do.)

Reply:
This is an important distinction, and not immediately apparent. Could you please add it to the comments in the header file -- whether each function can be called on the root, and how each function behaves with regards to missing intermediate subfields? Unless this is already explained in some other file.

Reply:
Done.

For `Set`
```
Returns a FieldValue with the field at the named path set to value.
Any absent parent of the field will also be created accordingly.
@param field_path The field path to set. Cannot be empty.
```
says it cannot be called on the root and missing intermediate subfield will be created.

For `Delete`
```
Returns a FieldValue with the field path deleted. If there is no field at
the specified path, the returned value is an identical copy.
@param field_path The field path to remove. Cannot be empty.
```
says it cannot be called on root and returns identical copy for missing intermediate subfields a.k.a. no field found at specified path.

For `Get`
```
Returns the value at the given path or absl::nullopt. If the path is empty,
an identical copy of the FieldValue is returned.
```
says it returns identical copy for root call and `nullopt` for missing intermediate subfields a.k.a. no field found at specified path.